testing/jormungandr-testing-utils/src/testing/node/grpc/server/builder.rs

```diff
@@ -76,19 +93,32 @@ fn start_thread(data: Arc<RwLock<MockServerData>>, mock_port: u16) -> MockContro
         log_file.path().to_string_lossy()
     );
 
-    let logger = MockLogger::new(log_file.path());
+    let (tx, rx) = sync_channel(100);
+    let logger = MockLogger::new(rx);
     let (shutdown_signal, rx) = oneshot::channel::<()>();
     let data_clone = data.clone();
 
-    tokio::spawn(async move {
-        let addr = format!("127.0.0.1:{}", mock_port);
-        let mock = JormungandrServerImpl::new(data_clone, log_file.path(), mock_port);
-
-        Server::builder()
-            .add_service(NodeServer::new(mock))
-            .serve_with_shutdown(addr.parse().unwrap(), rx.map(drop))
-            .await
-            .unwrap();
+    std::thread::spawn(move || {
+        let subscriber = tracing_subscriber::fmt()
+            .json()
+            .with_writer(move || ChannelWriter(tx.clone()))
+            .finish();
+        tracing::subscriber::with_default(subscriber, || {
+            let rt = tokio::runtime::Builder::new_current_thread()
+                .enable_all()
+                .build()
+                .unwrap();
+            rt.block_on(async move {
+                let addr = format!("127.0.0.1:{}", mock_port);
+                let mock = JormungandrServerImpl::new(data_clone, mock_port);
+                Server::builder()
+                    .add_service(NodeServer::new(mock))
+                    .serve_with_shutdown(addr.parse().unwrap(), rx.map(drop))
+                    .await
+                    .unwrap();
+            })
+        });
```

Comment:
Is this tracing subscriber for the tests or the node? I think I do not get what is being done here.

Reply:
This is the subscriber for the tests, here `Server` is a mock gRPC server that interacts with the node process. We set up  logs to track what actions the mock server performed.